app/src/main/java/com/newgogiyumandroid/MainActivity.kt

```diff
@@ -18,47 +29,61 @@ class MainActivity : AppCompatActivity() {
+            progressImg.visibility = View.GONE
+        }
     }
 
-    inner class NetworkThread: Thread() {
-        override fun run() {
-
-            val jsonArray = readData()
-
-            for (i in 0 until jsonArray.length()) {
-                val jsonObject = jsonArray.getJSONObject(i)
-
-                val imageUrl:String = jsonObject.getString("image")
-                val name = jsonObject.getString("name")
-                val ename = jsonObject.getString("e_name")
-
-                val foodList = FoodList(imageUrl, name, ename)
-                data.add(foodList)
+    fun readData() {
+        val client = OkHttpClient.Builder().build()
+        val req = Request.Builder().url("https://gogiyum.com/api/menu").build()
+        client.newCall(req).enqueue(object: Callback {
+            override fun onFailure(call: Call, e: IOException) {
+                // failure
             }
 
-            runOnUiThread {
-
-                // 2. 어뎁터 생성
-                val adapter = CustomAdapter()
-                // 3. 어뎁터에 데이터 전달
-                adapter.listData = data
-                // 4. 화면에 있는 리사이클러뷰에 어답터 연결 - recycler 은 recycler_view_example에 있는 recycler view  Id 임
-                recycler.adapter = adapter
-                // 5. 레이아웃 매니저 연결
-                recycler.layoutManager = LinearLayoutManager(applicationContext)
+            override fun onResponse(call: Call, response: Response) {
+                val launch = CoroutineScope(Dispatchers.Main).launch {
+
+                    val body = response.body!!.string()
+                    val jsonArray = JSONArray(body)
+
+                    for (i in 0 until jsonArray.length()) {
+                        val jsonObject = jsonArray.getJSONObject(i)
+
+                        val imageUrl: String = jsonObject.getString("image")
+                        val name = jsonObject.getString("name")
+                        val ename = jsonObject.getString("e_name")
+
+                        val foodList = FoodList(imageUrl, name, ename)
+                        data.add(foodList)
+                    }
+
+                    progressImg.visibility = View.GONE
+                    // 2. 어뎁터 생성
+                    val adapter = CustomAdapter()
+                    // 3. 어뎁터에 데이터 전달
+                    adapter.listData = data
+                    // 4. 화면에 있는 리사이클러뷰에 어답터 연결 - recycler 은 recycler_view_example에 있는 recycler view  Id 임
+                    recycler.adapter = adapter
+                    // 5. 레이아웃 매니저 연결
+                    recycler.layoutManager = LinearLayoutManager(applicationContext)
+                }
             }
-        }
-    }
 
-    // 음식 정보를 읽어와 JSONArray 로 반환하는 함수
-    fun readData(): JSONArray {
-        val url = URL("https://gogiyum.com/api/menu")
-        val connection = url.openConnection()
-        val data = connection.getInputStream().readBytes().toString(charset("UTF-8"))
-        return JSONArray(data)
+        })
     }
 
+    // 인터넷 연결 확인
```

Comment:
Why do we need to check internet connection?

Reply:
because if the internet is not connected, I can't get data from server, so I need to show other view.

Reply:
Sounds good.